pkg/util/goroutinemap/common/exponential_backoff.go

```diff
@@ -0,0 +1,77 @@
+/*
```

Comment:
@saad-ali This is a neat piece of code. I would just move it to pkt/util and out of pkg/util/goroutinemap. Share it with the world of kubernetes.